src/SecurityManager.js

```diff
@@ -272,6 +348,57 @@ export async function accessSecretStorage(func = async () => { }, forceReset = f
         secretStorageBeingAccessed = false;
         if (!isCachingAllowed()) {
             secretStorageKeys = {};
+            secretStorageKeyInfo = {};
+        }
+    }
+}
+
+// FIXME: this function name is a bit of a mouthful
+export async function tryToUnlockSecretStorageWithDehydrationKey(client) {
```

Comment:
This function should exit early if the feature flag is disabled.

Reply:
`dehydrationCache.key` will only be set if they had a previously dehydrated device, and they've already been prompted to dehydrate it, so the user has already opted in to use dehydration.

Reply:
I've added the feature flag check in the block where it tries to rehydrate the device.  `dehydrationCache.key` will never be set if that part never runs, so that will effectively make this exit early, so I don't think we need an additional feature flag check here.